common/web/types/tests/kmx/kmx-plus/element-string.tests.ts

```diff
@@ -0,0 +1,480 @@
+        ];
+        assert.deepEqual(actual, expected);
+      });
+      it('can apply single prebase to all', () => {
+        sections.strs.allocString = stubStrsAllocString_Char;
+        const actual = ElementString.fromStrings(
+          sections,
+          "𐌰𐌱𐌲",
+          null,
+          null,
+          null,
+          "1",
+        );
+        const expected = [
+          initElemElement(GOTHIC_A, undefined, 0, 0, ElemElementFlags.prebase),
+          initElemElement(GOTHIC_B, undefined, 0, 0, ElemElementFlags.prebase),
+          initElemElement(GOTHIC_C, undefined, 0, 0, ElemElementFlags.prebase),
+        ];
+        assert.deepEqual(actual, expected);
+      });
+      it('can handle prebase string that is too short', () => {
+        sections.strs.allocString = stubStrsAllocString_Char;
+        const actual = ElementString.fromStrings(
+          sections,
+          "𐌰𐌱𐌲",
+          null,
+          null,
+          null,
+          "1 0",
+        );
+        const expected = [
+          initElemElement(GOTHIC_A, undefined, 0, 0, ElemElementFlags.prebase),
+          initElemElement(GOTHIC_B, undefined, 0, 0, ElemElementFlags.none),
+          initElemElement(GOTHIC_C, undefined, 0, 0, ElemElementFlags.none),
+        ];
+        assert.deepEqual(actual, expected);
+      });
+      it('can create an ElementString from a uset string', () => {
+        ElementParser.segment = stubElementParserSegment_Uset;
+        sections.strs.allocString = stubStrsAllocString_Str;
+        sections.uset.allocUset = stubUsetAllocUset;
+        const actual = ElementString.fromStrings(sections, "[𐌰-𐍊]");
+        const expected = [
+          initElemElement(
+            new StrsItem(''),
+            GOTHIC_USETITEM,
+            0,
+            0,
+            constants.elem_flags_type_uset,
+          ),
+        ];
+        assert.deepEqual(actual, expected);
+      });
+      it('returns null for an invalid unicode set size', () => {
+        ElementParser.segment = stubElementParserSegment_Uset;
+        sections.usetparser.sizeUnicodeSet = (pattern: string) : number => -1;
+        assert.isNull(ElementString.fromStrings(sections, "[𐌰-𐍊]"));
+      });
+      it('returns null if it cannot parse the unicode set', () => {
+        ElementParser.segment = stubElementParserSegment_Uset;
+        sections.usetparser.parseUnicodeSet = (pattern: string, rangeCount: number) : UnicodeSet | null => null;
+        assert.isNull(ElementString.fromStrings(sections, "[𐌰-𐍊]"));
+      });
+      it('can handle quad strings', () => {
+        sections.strs.allocString = stubStrsAllocString_Char;
+        ElementParser.segment = stubElementParserSegment_Escaped;
+        const actual   = ElementString.fromStrings(sections, "\\ud800\\udf30");
+        const expected = [
+          initElemElement(HI_GOTHIC_A),
+          initElemElement(LO_GOTHIC_A),
+        ];
+        assert.deepEqual(actual, expected);
```

Comment:
this will work here, but is tested for in the keyboard compiler to be avoided ( the `\\ud800` and `\\udf30` will be rejected)